### PR TITLE
vpi: fix quickstart example for mcode build; add ref to CoCoTb-pygame demo

### PR DESCRIPTION
**Pull request by chuckb**
Running on Mac...installed ghdl via homebrew. That binary was build for mcode generation. Thus, this example did not work.

### CODE REVIEW

File: vpi/quickstart/run.sh [hunk 1]
```diff
@@ -22,4 +22,9 @@ if [ "$OS" = "Windows_NT" ]; then
 fi
 
 echo "> Execute tb"
-./tb --vpi=./vpi.vpi
+if [ `ghdl version | grep mcode | wc -l` = "1" ]; then
+  # if ghdl compiled with mcode, tb executable will not exist
+  ghdl -r tb --vpi=./vpi.vpi
```
Comment (umarcor) [inline vpi/quickstart/run.sh]: I believe it is also correct to use `ghdl -r` regardless of the backend. Hence, please remove the condition and leave line 27 only.

**Comment by chuckb** [comment]
I also added an example interactive GUI I whipped up by using cocotb and python.

**Comment by umarcor** [comment]
@chuckb would you mind partially rewriting the GHDL-CoCoTb example description to address the following details?

- Add "GUI" to the title. That's a very meaningful feature of your demo, because most GHDL + CoCoTb examples don't provide a GUI.
- I suggest adding cross-references: "*It provides a testbench and a matching GUI with I/O peripherals for testing a design, instead of examining waveforms; similar to the [GHDL VPI virtual board]() example or the [VGA (RGB image buffer)]() examples*".
- Mention "pygame". E.g. "*Using Python ([CoCoTb]() and [pygame](pygame.org)) makes interactive GUI creation more dynamic than using a compiled language (such as C++)*". Let's let users decide what is "easier" for them.

Moreover, if you want, your article might be contributed as a section to the docs of this repo, along with adding the sources as an example. I believe we might test it in CI with xvfb. That'd be in an upcoming PR, of course; no need to handle it now.

**Comment by chuckb** [comment]
Yes, I will make the changes you suggest. I would very much like to have the example included within the docs somehow, as there are not many GUI-related examples...none really with Cocotb as you mentioned. Fusing Cocotb and pygame was an interesting exercise and kind of fun.

 I should have split up the PR, I recognize...

**Comment by umarcor** [comment]
I did a minor style fix, I rebased and I merged. Thanks!

>  would very much like to have the example included within the docs somehow, as there are not many GUI-related examples...none really with Cocotb as you mentioned.

You can start creating subdir `vpi/cocotb/pygame` in a branch, and adding the sources of the example there. Then, we can iterate on that to add it to CI and to the docs.

BTW, https://twitter.com/unaimarcor/status/1476772035728715782

**Comment by chuckb** [comment]
Thanks for the kind mention. I also integrated this example into a FuseSoC core...lots of open source goodies to explore. https://github.com/chuckb/alchitry-au-fusesoc

**Pull request merged**